app/stac_api/models.py

```diff
@@ -236,37 +238,50 @@ def update_bbox_extent(self, action, item_geom, item_id):
-                self.extent_geometry = Polygon.from_bbox(GEOSGeometry(item_geom).extent)
-            # we need to iterate trough the items
-            else:
+        try:
+            # insert (as item_id is None)
+            if action == 'up' and item_id is None:
+                # the first item of this collection
+                if self.extent_geometry is None:
+                    self.extent_geometry = Polygon.from_bbox(GEOSGeometry(item_geom).extent)
+                # there is already a geometry in the collection a union of the geometries
+                else:
+                    self.extent_geometry = Polygon.from_bbox(
+                        GEOSGeometry(self.extent_geometry).union(GEOSGeometry(item_geom)).extent
+                    )
+
+            # update
+            if action == 'up' and item_id:
+                # is the new bbox larger than (and covering) the existing
+                if Polygon.from_bbox(GEOSGeometry(item_geom).extent).covers(self.extent_geometry):
+                    self.extent_geometry = Polygon.from_bbox(GEOSGeometry(item_geom).extent)
+                # we need to iterate trough the items
+                else:
+                    qs = Item.objects.filter(collection_id=self.pk).exclude(id=item_id)
+                    union_geometry = GEOSGeometry(item_geom)
+                    for item in qs:
+                        union_geometry = union_geometry.union(item.geometry)
+                    self.extent_geometry = Polygon.from_bbox(union_geometry.extent)
+
+            # delete, we need to iterate trough the items
+            if action == 'rm':
                 qs = Item.objects.filter(collection_id=self.pk).exclude(id=item_id)
-                union_geometry = GEOSGeometry(item_geom)
+                union_geometry = GEOSGeometry('Multipolygon EMPTY')
                 for item in qs:
                     union_geometry = union_geometry.union(item.geometry)
                 self.extent_geometry = Polygon.from_bbox(union_geometry.extent)
-
-        # delete, we need to iterate trough the items
-        if action == 'rm':
-            qs = Item.objects.filter(collection_id=self.pk).exclude(id=item_id)
-            union_geometry = GEOSGeometry('Multipolygon EMPTY')
-            for item in qs:
-                union_geometry = union_geometry.union(item.geometry)
-            self.extent_geometry = Polygon.from_bbox(union_geometry.extent)
+        except GEOSException as error:
+            logger.error(
+                'Failed to update spatial extend in collection %s with item %s action=%s: %s',
+                self.collection_name,
+                item_name,
+                action,
+                error
+            )
+            raise GEOSException(
+                f'Failed to update spatial extend in colletion {self.collection_name} with item '
+                f'{item_name}: {error}'
+            )
```

Comment:
I usually prefer to just re-raise the original exception to keep the stacktrace with the error:
```
except GEOSException as error:
    logger.error(...)
    raise
```

Reply:
Well the stack trace about is kept as well as the error, the error message is just enhanced to give a hint to the user what is failing otherwise the user doesn't have a clue about the error, the orignal GEOSException message doesn't help at all that's why I completed it with useful infos.